pkg/controller/nginxingress/nginx_ingress_controller.go

```diff
@@ -141,7 +142,7 @@ func (n *nginxIngressControllerReconciler) Reconcile(ctx context.Context, req ct
 		n.updateStatus(&nginxIngressController, controllerDeployment, ingressClass, managedRes, collisionCountErr)
 		if statusErr := n.client.Status().Update(ctx, &nginxIngressController); statusErr != nil {
 			lgr.Error(statusErr, "unable to update NginxIngressController status")
-			if err == nil {
+			if err == nil && !k8serrors.IsConflict(statusErr) {
```

Comment:
if there's a conflict we do want to exit. We should just call with immediate requeue


Also why is there a conflict tho? That's the real question. My guess is the cache